init/cli/start.go

```diff
@@ -29,59 +28,35 @@ func startCommand() cli.Command {
 		Name: "start",
 		Usage: `
-Runs the command defined by the given static and custom configurations and stores the PID of the resulting process in
-the given pid file.
-`,
-		Flags: []flag.Flag{
-			flag.StringFlag{
-				Name:  launcherStaticFileParameter,
-				Value: "service/bin/launcher-static.yml",
-				Usage: "The location of the LauncherStatic file configuration of the started command"},
-			flag.StringFlag{
-				Name:  launcherCustomFileParameter,
-				Value: "var/conf/launcher-custom.yml",
-				Usage: "The location of the LauncherCustom file configuration of the started command"},
-			flag.StringFlag{
-				Name:  pidfileParameter,
-				Value: "var/run/service.pid",
-				Usage: "The location of the file storing the process ID of the started command"},
-			flag.StringFlag{
-				Name:  outFileParameter,
-				Value: "var/log/startup.log",
-				Usage: "The location of the file to which STDOUT and STDERR of the started command are redirected"},
+Launches the process defined by the static and custom configurations at service/bin/launcher-static.yml and
+var/conf/launcher-custom.yml. Writes its PID to var/run/service.pid and redirects its output to var/log/startup.log. If
+successful, exits 0, otherwise exits 1 and writes an error message to stderr.`,
+		Action: func(_ cli.Context) error {
+			return start()
 		},
-		Action: doStart,
 	}
 }
 
-func doStart(ctx cli.Context) error {
-	launcherStaticFile := ctx.String(launcherStaticFileParameter)
-	launcherCustomFile := ctx.String(launcherCustomFileParameter)
-	pidfile := ctx.String(pidfileParameter)
-	stdoutFileName := ctx.String(outFileParameter)
+func start() error {
+	if _, _, err := lib.GetProcessStatus(); err == nil {
+		// Process already running, don't restart it.
+		return nil
+	}
 
-	stdoutFile, err := os.Create(stdoutFileName)
+	outputFile, err := os.Create(lib.OutputFile)
 	if err != nil {
-		msg := fmt.Sprintln("Failed to create startup log file", err)
-		return respondError(msg, err, 1)
+		return cli.WithExitCode(1, errors.Wrap(err, "failed to create startup log file"))
 	}
 
-	originalStdout := os.Stdout
-	os.Stdout = stdoutFile // log command assembly output to file instead of stdout
-	defer func() {
-		os.Stdout = originalStdout
-	}()
-	cmd, err := launchlib.CompileCmdFromConfigFiles(launcherStaticFile, launcherCustomFile)
+	cmd, err := launchlib.CompileCmdFromConfigFiles(lib.LauncherStaticFile, lib.LauncherCustomFile, outputFile)
 	if err != nil {
-		msg := fmt.Sprintln("Failed to assemble Command object from static and custom configuration files", err)
-		return respondError(msg, err, 1)
+		return cli.WithExitCode(1,
+			errors.Wrap(err, "failed to assemble command from static and custom configuration files"))
 	}
 
-	pid, err := lib.StartCommandWithOutputRedirectionAndPidFile(cmd, stdoutFile, pidfile)
-	if err != nil {
-		msg := fmt.Sprintln("Failed to start process", err)
-		return respondError(msg, err, 1)
+	if err = lib.StartCommand(cmd, outputFile); err != nil {
```

Comment:
Minor, but when using this style, do a `err :=` assignment -- this ensures that the variable is scoped just to this block of code (and doesn't update the value of the outer `err` variable)